app/src/main/java/crux/bphc/cms/helper/CourseDataHandler.java

```diff
@@ -79,8 +79,10 @@ public String getActionBarTitle(int courseId){
             if (realm.where(Course.class).equalTo("id", course.getId()).findFirst() == null) {
                 newCourses.add(course);
             }
+            else
+                course.setFavoriteStatus(realm.where(Course.class).equalTo("id", course.getId()).findFirst().isFavorite());
```

Comment:
Consider retrieving and caching course ids of all courses marked as favorite. This will prevent a lookup on each iteration.